### ISSUE TITLE
Accessing C# internal set with public get results in no Property Initialization, and odd setter acce

### ISSUE THREAD

**Issue by ImaginaryDevelopment** [dotnet/fsharp#7165]
Please provide a succinct description of the issue.

**Repro steps**

Provide the steps required to reproduce the problem:

1. `dotnet new console -lang F#`
2. try to access a property declared as such `public int Prop { internal get; set; }` or try creating an instance of that class and set the property in property initialization syntax.

If possible attach a zip file with the repro case. This often makes it easier for others to reproduce.
The zip file should ideally represent the situation just before the call/step that is problematic.
repro repo resides radically realistically relativistically responsive, really https://github.com/tldrlol/Fs-Prop-Oddity/blob/master/DsTest.Cs/C.cs

**Expected behavior**

`let x = C(Prop=1)` or
`let x = C(); x.Prop <- 1` works

**Actual behavior**

only `let x = C(); x.set_Prop 1` works

**Known workarounds**

`x.set_Prop`

**Comment by cartermp** [comment]
Found in https://github.com/333fred/OrannisBot

**Comment by TIHan** [comment]
I agree. I would definitely expect to use the property in initialization syntax and setter as described.

I don't think this is a bug or regression though, but rather a feature improvement.

**Comment by TIHan** [comment]
In `AccessibilityLogic.fs`:
```fsharp
let GetILAccessOfILPropInfo (ILPropInfo(tinfo, pdef)) =
    let tdef = tinfo.RawMetadata
    let ilAccess =
        match pdef.GetMethod with 
        | Some mref -> (resolveILMethodRef tdef mref).Access 
        | None -> 
            match pdef.SetMethod with 
            | None -> ILMemberAccess.Public
            | Some mref -> (resolveILMethodRef tdef mref).Access
    ilAccess
```

For IL defined properties, access of a property is defined by the getter method first. If there is no getter, then it will be defined by the setter method. Because our C#/IL property has a getter method, but is marked internal, then this code in `InfoReader.fs`:
```fsharp
    let pinfos = pinfos |> List.filter (IsPropInfoAccessible g amap m ad)
    pinfos
```
Will filter out those properties. I'd argue this is a bug and probably not intended.

Because this changes the language, we do need a RFC.

**Comment by Lamarth** [comment]
Same deal with private instead of internal.

I almost raised a new issue, but related issues found this!
